torch/csrc/dynamo/eval_frame.c

```diff
@@ -509,7 +510,7 @@ static PyObject* _custom_eval_frame_shim(
   //  - Python callable(): enables TorchDynamo
   PyObject* callback = eval_frame_callback_get();
 
-  if (callback == Py_None) {
+  if (!eval_frame_callback_enabled || callback == Py_None) {
     return eval_frame_default(tstate, frame, throw_flag);
```

Comment:
Would it make more sense for eval_frame_callback_get to consult the enable/disable status?

Reply:
I want to keep `eval_frame_callback_enabled` and `eval_frame_callback_get` independent. `eval_frame_callback_get` should return the callback dynamo _wants_ to use, `eval_frame_callback_enabled` determines if we actually use the callback or the default.

In particular, `set_eval_frame_py` (used in Dynamo convert_frame) retrieves the previous set callback via `eval_frame_callback_get()`, and I don't want `eval_frame_callback_enabled` getting in the way.